packages/dcos-integration-test/extra/test_dcos_log.py

```diff
@@ -236,3 +236,104 @@ def get_task_url(dcos_api_session, task_name, stream=False):
     return '/system/v1/agent/{}/logs/v1/{}/framework/{}/executor/{}/container/{}'.format(slave_id, endpoint_type,
                                                                                          framework_id, executor_id,
                                                                                          container_id)
+
+
+def test_log_v2_text(dcos_api_session):
+    for node in dcos_api_session.masters + dcos_api_session.all_slaves:
+        response = dcos_api_session.logs.get('v2/component?limit=10', node=node)
+        check_response_ok(response, {'Content-Type': 'text/plain'})
+
+        # expect 10 lines
+        lines = list(filter(lambda x: x != '', response.content.decode().split('\n')))
+        assert len(lines) == 10, 'Expect 10 log entries. Got {}. All lines {}'.format(len(lines), lines)
+
+
+def test_log_v2_server_sent_events(dcos_api_session):
+    for node in dcos_api_session.masters + dcos_api_session.all_slaves:
+        response = dcos_api_session.logs.get('v2/component?limit=1', node=node, headers={'Accept': 'text/event-stream'},
+                                             stream=True)
+        check_response_ok(response, {'Content-Type': 'text/event-stream', 'Cache-Control': 'no-cache'})
+        lines = response.iter_lines()
+        sse_id = next(lines)
+        assert sse_id, 'First line must be id. Got {}'.format(sse_id)
+        data = next(lines).decode('utf-8', 'ignore')
+        validate_sse_entry(data)
+
+
+def test_log_v2_stream(dcos_api_session):
+    for node in dcos_api_session.masters + dcos_api_session.all_slaves:
+        response = dcos_api_session.logs.get('v2/component?skip=-1', node=node, stream=True,
+                                             headers={'Accept': 'text/event-stream'})
+        check_response_ok(response, {'Content-Type': 'text/event-stream', 'Cache-Control': 'no-cache'})
+        lines = response.iter_lines()
+        sse_id = next(lines)
+        assert sse_id, 'First line must be id. Got {}'.format(sse_id)
+        data = next(lines).decode('utf-8', 'ignore')
+        validate_sse_entry(data)
+
+
+def test_log_v2_proxy(dcos_api_session):
+    r = dcos_api_session.get('/mesos/master/slaves')
+    check_response_ok(r, {})
+
+    data = r.json()
+    slaves_ids = sorted(x['id'] for x in data['slaves'] if x['hostname'] in dcos_api_session.all_slaves)
+
+    for slave_id in slaves_ids:
+        response = dcos_api_session.get('/system/v1/agent/{}/logs/v2/component?skip=-10&limit=10'.format(slave_id))
```

Comment:
FYI you can leave off the `?` and `&` in your query strings and instead use the query parameter of this object which takes a simple dict. Just a stylistic choice, so whatever works for you is good
https://github.com/dcos/dcos-test-utils/blob/master/dcos_test_utils/helpers.py#L103